src/SshProcessConnector.php

```diff
@@ -66,7 +66,27 @@ public function connect($uri)
             return \React\Promise\reject(new \InvalidArgumentException('Invalid target URI'));
         }
 
-        $process = new Process($this->cmd . ' -W ' . \escapeshellarg($parts['host'] . ':' . $parts['port']));
+        $command = $this->cmd . ' -W ' . \escapeshellarg($parts['host'] . ':' . $parts['port']);
+
+        // try to get list of all open FDs (Linux only) or simply assume range 3-1024 (FD_SETSIZE)
+        $fds = @scandir('/proc/self/fd');
```

Comment:
FYI working on a package to do this cross platform: https://github.com/WyriHaximus/php-file-descriptors

Reply:
@WyriHaximus Thank you, will definitely keep an eye on this! :+1:

Reply:
@clue currently onto the daunting task of figuring it out for `Windows`...